src/dbnode/client/session.go

```diff
@@ -114,6 +114,10 @@ var (
 	errUnableToEncodeTags = errors.New("unable to include tags")
 	// errEnqueueChIsClosed is returned when attempting to use a closed enqueuCh.
 	errEnqueueChIsClosed = errors.New("error enqueueCh is cosed")
+
+	defaultCheckedBytesPoolBucketSizes = []pool.Bucket{
+		{Capacity: 128, Count: 4096},
```

Comment:
I've chosen the numbers pretty randomly. Any suggestions for more suitable values?

Reply:
Have you explored reusing some global pool by passing it to `newSession` via `Options` (like it is done with eg. `IdentifierPool`, `ContextPool`)? If we could do it, we would not have to solve this question.

Reply:
Hm, I didn't find any suitable global pools (there's one in being placed in [`storage.Options`](https://github.com/m3db/m3/blob/e44d247f803bb74240392230ba39c73db7553e9e/src/dbnode/server/server.go#L1551-L1556), but I doubt we could use it for client)

Reply:
For our use cases, annotations will be just few bytes long. So I suggest adding something like 16 byte capacity bucket for that. And maybe you can keep the bigger one, just in case.